### ISSUE TITLE
Fuzzy matching bug with Flash files

### ISSUE THREAD

**Issue by despens** [webrecorder/pywb#343]
This happens with pywb version 2.0.3 installed via pip on Python 3.6.5:

Pywb returns wrong data for certain URL requests, perhaps related to `.swf` (Flash) files.

To reproduce:

## Create WARC file

Directory structure:
```
files
├── path1
│   └── stuff
│       └── deep-down.html
└── path2
    ├── deep-down
    │   ├── deep-down.html
    │   ├── image.png
    │   └── what.swf
    └── index.html
```

The files are all have arbitrary contents.

Create a WARC file from this:

```bash
$ warcit -o http://www.website.co.br/~dinosaurs/t/ files -v
[DEBUG] Writing "http://www.website.co.br/~dinosaurs/t/path1/stuff/deep-down.html" (text/html) @ "2018-06-13T08:25:37Z" from "files/path1/stuff/deep-down.html"
[DEBUG] Writing "http://www.website.co.br/~dinosaurs/t/path2/index.html" (text/html) @ "2018-06-13T08:29:18Z" from "files/path2/index.html"
[DEBUG] Adding auto-index: http://www.website.co.br/~dinosaurs/t/path2/ -> http://www.website.co.br/~dinosaurs/t/path2/index.html
[DEBUG] Writing "http://www.website.co.br/~dinosaurs/t/path2/deep-down/image.png" (image/png) @ "2018-06-12T14:05:18Z" from "files/path2/deep-down/image.png"
[DEBUG] Writing "http://www.website.co.br/~dinosaurs/t/path2/deep-down/deep-down.html" (text/html) @ "2018-06-13T08:27:30Z" from "files/path2/deep-down/deep-down.html"
[DEBUG] Writing "http://www.website.co.br/~dinosaurs/t/path2/deep-down/what.swf" (application/x-shockwave-flash) @ "2018-06-13T08:39:31Z" from "files/path2/deep-down/what.swf"
[INFO] Wrote 6 resources to files.warc.gz
```

## Add to pywb

```bash
wb-manager add test files.warc.gz
wayback
```

## Access on localhost:

Requesting a path that partly matches that of any resource in the WARC will result in the flash file being served instead of a not found error.

![selection_999 725](https://user-images.githubusercontent.com/571494/41340801-7e31291c-6ef8-11e8-8c9b-e57ac82d76a0.png)

![selection_999 726](https://user-images.githubusercontent.com/571494/41340807-83cbb9f0-6ef8-11e8-93c1-d03d0815d871.png)

![selection_999 727](https://user-images.githubusercontent.com/571494/41340817-897c2f1a-6ef8-11e8-8e68-d2be30c2bc47.png)

Only when a path is requested that doesn't match parts of resource URLs in the WARC, an error is served:

![selection_999 729](https://user-images.githubusercontent.com/571494/41340984-e9a0b76c-6ef8-11e8-863d-e1e4008bbdab.png)

## Data

[WARC file for testing](https://github.com/webrecorder/pywb/files/2097408/files.warc.gz)

This is the isolation of many issues encountered during several projects created with warcit.

**Comment by despens** [comment]
This behavior persisted with pywb2.0.4; essentially random flash files are displayed instead of 404 errors from pywb in numerous restoration projects.

**Comment by ikreymer** [comment]
Should be fixed in 2.1.0!
